OpenRA.Game/Scripting/ScriptObjectWrapper.cs

```diff
@@ -63,9 +67,22 @@ protected void Bind(object[] clrObjects)
 
 		protected void Unbind(Type targetType)
 		{
+#if NET5_0_OR_GREATER
+			// NOTE: In newer versions of .NET modifying the collection by calling Remove while iterating over it is valid
 			foreach (var m in members)
 				if (targetType == m.Value.Target.GetType())
 					members.Remove(m.Key);
+#else
+			// PERF: Re-use instead of allocating a new list on each unbind
+			membersToRemove.Clear();
+
+			foreach (var m in members)
+				if (targetType == m.Value.Target.GetType())
+					membersToRemove.Add(m.Key);
+
+			foreach (var m in membersToRemove)
+				members.Remove(m);
```

Comment:
You could use linq. members.Where.ToArray and then itterate over this array

Reply:
That would mean we'd still allocate a new array on each unbind. 🤔